### ISSUE TITLE
How to run the demo app?

### ISSUE THREAD

**Issue by YoungxHelsinki** [mapbox/mapbox-gl-native#5229]
**Platform: iOS**
**Mapbox SDK version: `commit 623da8bfeb7adcbf1774a14c9a0dc46469891cef`**

Hi all,

I would like to contribute to Mapbox ios SDK and I tried to run the demo app inside
the mapbox-gl-native. How I ever I got multiple errors. The first one was about 
SMCalloutView.m/h missing. I figured that the SMCalloutView directory was linked 
to [other repo](https://github.com/nfarina/calloutview/tree/2aede5d8d1577101bf18405246220e7a710df607). 
I copied the corresponding files to the "right" path, i.e., 
`/mapbox-gl-native/platform/ios/vendor/SMCalloutView/`

Then I got another error:
`The file “mbgl.xcconfig” couldn’t be opened because there is no such file. (/mapbox-gl-native/build/ios/mbgl.xcconfig)`
![screen shot 2016-06-03 at 17 24 31](https://cloud.githubusercontent.com/assets/5738363/15781744/3653e5c4-29b0-11e6-9478-553ebfa99b99.png)
This time I couldn't get any clue about what that file is.

### In short:
1. How can I resolve all those red things in my workspace?
2. Why are they deliberately missing from the repo?

Thanks.

**Comment by friedbunny** [comment]
Thanks for digging in! Our [developing instructions](https://github.com/mapbox/mapbox-gl-native/blob/e9dcfa0ed627ecb8783d1ac7f38a24db51cf8cb8/platform/ios/DEVELOPING.md) appear to be lacking an important setup step:

```
git submodule init
git submodule update
```

This will get you SMCalloutView and KIF, which you shouldn’t install manually. I’ll add this omitted step to our instructions.

As for the other red files, these are build products that will be created when you build the various app and test targets. The `iosapp` target is the demo app that you’ll likely be most interested in.

**Comment by YoungxHelsinki** [comment]
Thanks @friedbunny for the reply. Actually I did learn about the submodule thing on irc channel but it didn't solve the issue. I still get this error:
`/mapbox-gl-native/include/mbgl/util/geo.hpp:6:10: 'mapbox/geometry/point.hpp' file not found`
![screen shot 2016-06-03 at 18 16 50](https://cloud.githubusercontent.com/assets/5738363/15783618/9392a52a-29b7-11e6-9463-33e5705df9c2.png)

**Comment by jfirebaugh** [comment]
It's not necessary to run any `git submodule` commands manually. From a clean checkout, the _only_ command that's necessary to create a buildable Xcode project is:

```
make xproj
```

@YoungxHelsinki -- if, from a fresh checkout, that command doesn't work for you, please post full console and Xcode output.

**Comment by friedbunny** [comment]
@jfirebaugh On my machine, `make xproj` (or `make iproj`, for that matter) will not work until the mason submodule is checked out and it does not do this automatically.

```sh
$ make xproj
./configure platform/osx/scripts/configure.sh build/osx/config.gypi osx
* Using Python 2.7.10
./configure: line 42: mason: command not found
make: *** [build/osx/config.gypi] Error 127
```

After mason exists, `make iproj` is sufficient to create a functional workspace.

**Comment by jfirebaugh** [comment]
@friedbunny Indeed -- fixed in a758b41b06d21b74c52dd4e1b001eb6e1a82f913.

**Comment by YoungxHelsinki** [comment]
@friedbunny & @jfirebaugh  -- Works! Thank you for the support!!! 👍

**Comment by hashemp206** [comment]
I can'r run sample codes! after cloning and  entering `make iproj` the workspace is look like this: 
<img width="197" alt="screen shot 2016-11-02 at 4 33 30 pm" src="https://cloud.githubusercontent.com/assets/3206748/19929746/20067700-a11a-11e6-8929-e3f5f26af46a.png">

how can I run the sample project?
